components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -554,19 +623,51 @@ TEST_F(SolanaTxManagerUnitTest, AddAndApproveTransaction) {
       std::vector<std::string>(
           {from_account_address, to_account, mojom::kSolanaSystemProgramId}),
       std::vector<mojom::SolanaMessageAddressTableLookupPtr>(), nullptr,
-      nullptr);
+      nullptr, nullptr);
 
+  // First add a partially signed transaction - it should not fetch add any
+  // priority fee instructions or add a fee estimate.
+  auto param = mojom::SolanaSignTransactionParam::New(
+      "test", std::vector<mojom::SignaturePubkeyPairPtr>());
+  param->signatures.emplace_back(mojom::SignaturePubkeyPair::New(
+      std::vector<uint8_t>(kSolanaSignatureSize, 1), sol_account()->address));
+  solana_tx_data->sign_transaction_param = std::move(param);
   auto tx = SolanaTransaction::FromSolanaTxData(solana_tx_data.Clone());
   ASSERT_TRUE(tx);
-
   std::string meta_id1;
   AddUnapprovedTransaction(mojom::kSolanaMainnet, solana_tx_data.Clone(),
                            from_account, &meta_id1);
-
   auto tx_meta1 = solana_tx_manager()->GetTxForTesting(meta_id1);
+  EXPECT_EQ(*tx_meta1->tx(), *tx);
+
+  // Remove partial signature
+  solana_tx_data->sign_transaction_param = nullptr;
+  tx = SolanaTransaction::FromSolanaTxData(solana_tx_data.Clone());
+  ASSERT_TRUE(tx);
+
+  SetInterceptor(latest_blockhash1_, last_valid_block_height1_, tx_hash1_, R"({
+      "jsonrpc":"2.0","id":1,
+      "result": {
+        "context":{"slot":123065869},
+        "value": 5000
+      }
+    })",
```

Comment:
This is copied many times, please define a constant in this file, maybe kMockTxBaseFeeRsp. 
I'm not 100% sure it's for tx base fee btw, it's kind of hard to tell.